pkg/crd/schema_visitor.go

```diff
@@ -48,11 +48,18 @@ type schemaWalker struct {
 // visitor (this is as simple as passing a pointer in most cases,
 // but special care needs to be taken to persist with maps).
 func (w schemaWalker) walkSchema(schema *apiext.JSONSchemaProps) {
-	subVisitor := w.visitor.Visit(schema)
-	if subVisitor == nil {
-		return
+	// Walk a potential chain of schema references
+	subVisitor := w.visitor
+	for {
+		subVisitor = subVisitor.Visit(schema)
+		if subVisitor == nil {
+			return
```

Comment:
can you add a test for this?

Make sure to have one for circular references too

Reply:
Added a test that on current master produces

```
    typeRefChain:
      $ref: '#/definitions/testdata.kubebuilder.io~1cronjob~1api~0Integer'
```

but works correctly with this PR.

Also added cycle detection logic so that this `for` terminates for circular references.
I couldn't come up with circular test that would compile though.

Reply:
Found a simpler reproducer from #324 and switched to using it instead.